nilearn/maskers/nifti_maps_masker.py

```diff
@@ -229,7 +229,11 @@ def generate_report(self, displayed_maps=10):
         from nilearn.reporting.html_report import generate_report
 
         if (
-            displayed_maps != "all"
+            (isinstance(displayed_maps, str) and displayed_maps != "all")
+            or (
+                not isinstance(displayed_maps, str)
+                and np.all(displayed_maps != "all")
+            )
```

Comment:
The logic here is abit complex. Cna you give plain word explanation of the intended behevior

Reply:
Definitely clearer. To me at least.